src/components/ChatBot/assistant.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import { GPTMessage, GPTMessageSchema } from "@/lib/chatbot/validators";
+
+import Bubble from "./bubble";
+import Message from "./message";
+
+import styles from "./assistant.module.css";
+
+function Assistant() {
+  const [chatOpen, setChatOpen] = useState(false);
+  const [userText, setUserText] = useState("");
+  const [messages, setMessages] = useState<GPTMessage[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const messagesRef = useRef<HTMLDivElement | null>(null);
+
+  /* Calls the server for GPT response */
+  const getMessage = async (payload: GPTMessage[]): Promise<GPTMessage> => {
+    const response = await fetch("http://localhost:3000/api/chatbot/message", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json"
+      },
+      body: JSON.stringify(payload)
+    });
+    const data = await response.json();
+    const reply = GPTMessageSchema.safeParse(data);
+    if (reply.success) return reply.data;
+    return { role: "system", content: "Sorry, please try again" };
+  };
+
+  /* Get GPT initial intro message */
+  useEffect(() => {
+    const getInitialMessage = async () => {
+      setIsLoading(true);
+      const initialMessage = await getMessage([]);
```

Comment:
No, so the way the back end works is the assistant requires a prompt. Having the front end send an empty array lets the back end know the user has opened the chat window. The back end will then prepend the prompt and send it to gpt and then respond to the front end with the assistant's initial message.

I feel this is a better approach over hard coding an initial message given we have to prompt the assistant anyway, so may aswell return the initial response rather than hard code it. If you have an api key, you can test this by adding it to your env and running the project. When you open the chat window, the very first message that appears is a result of this code.

Reply:
Yeah that makes sense for the frontend, might be better to wait for the initial user message before querying GPT in the backend though.